### PR TITLE
Feature/dashboard

### PR DESCRIPTION
**Pull request by louismunro**
# Description

Adds a performance dashboard to PacketFence allowing tracking of metrics such as radius requests, number of apache processes, time to serve an aaa request etc.
# Impacts

Adds six new services:
carbon-cache
carbon-relay
httpd.graphite
statsd
collectd
radsniff3

Also changes significantly the appearance of the dashboard in the admin GUI.
# Delete branch after merge

YES 
# NEWS file entries
- New performance dashboard based on Graphite allows tracking of core performance metrics such as number and latency of radius requests, number of httpd.aaa processes and authorization latency.

### CODE REVIEW

**Comment by louismunro** [comment]
Please review me!

File: html/pfappserver/lib/pfappserver/Model/DB.pm [hunk 1]
```diff
@@ -58,7 +58,26 @@ sub assign {
...
+    $dbHandler->do("FLUSH PRIVILEGES");
+    if ( $DBI::errstr ) {
+        $status_msg = ["Error creating the user [_1] on database [_2]",$user,$db];
+        $logger->warn("$DBI::errstr");
+        return ( $STATUS::INTERNAL_SERVER_ERROR, $status_msg );
+    }
+    $status_msg = ["Successfully created the user [_1] on database [_2]",$user,$db];
 
+    # Create pf_graphite database
+    $db = $dbHandler->quote_identifier("${db}_graphite");
```
Comment (jrouzierinverse) [inline html/pfappserver/lib/pfappserver/Model/DB.pm:71]: In the upgrade schema the database is hardcoded to pf_graphite.

File: html/pfappserver/lib/pfappserver/PacketFence/Controller/Graph.pm [hunk 2]
```diff
@@ -63,7 +67,7 @@ Save the period range for a specific section.
 sub _saveRange :Private {
     my ($self, $c, $section, $start, $end) = @_;
 
-    if ($start && $end) {
+    if (length $start && length $end) {
```
Comment (jrouzierinverse) [inline html/pfappserver/lib/pfappserver/PacketFence/Controller/Graph.pm]: also check if defined?

File: lib/pf/StatsD.pm [hunk 3]
```diff
@@ -38,6 +39,7 @@ sub new {
     $host = 'localhost' unless defined $host;
     $port = 8125        unless defined $port;
     my $hostname = hostname; 
+    $hostname =~ s/\./_/g; # replace dots with underscores
```
Comment (jrouzierinverse) [inline lib/pf/StatsD.pm]: Why?

Comment (louismunro) [inline lib/pf/StatsD.pm]: Because it's used as the statsd delimiter already.
I'll make this more obvious.

**Pull request merged**